Doberman/Reading.py

```diff
@@ -115,6 +116,22 @@ def more_processing(self, value):
             value = int(value)
         return value
 
+    def sensiblesigfigs(self, reading, lowlim, upplim, defaultsigfigs=3):
+        """
+        Rounds reading to a sensible number of significant figures.
+
+        In general rounds to defaultsigfigs significant figures.
+        If the lowlim and upplim are rather close, have at least
+        one more than the number of decimal places to distinguish
+        them. For example: with limits 1.023 and 1.044, readings have
+        three decimal places.
+        """
+        mindps = 1 - floor(log10(upplim - lowlim))
+        minsfs = floor(log10(reading)) + 1 + mindps
+        sfs = max(minsfs, 3)
+        print(minsfs, sfs)
+        return f'{reading:.{sfs}g}'
```

Comment:
I had no idea you could nest things inside f-strings like this

Reply:
Me neither until today